test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -80,11 +83,22 @@ class AndroidArgs(
 
         val filteredTests = getTestMethods(testLocalApk)
 
-        calculateShards(
-            filteredTests,
-            testTargetsAlwaysRun,
-            testShards
-            )
+        if (true) {
+            val oldTestResult = GcStorage.downloadJunitXml(this) ?: JUnitTestResult(mutableListOf())
+
+            val start = System.nanoTime()
+            val result = convertShards(Shard.calculateShardsByTime(filteredTests, oldTestResult, testShards))
+            val shardCreationTime = TimeUnit.SECONDS.convert(System.nanoTime() - start, TimeUnit.NANOSECONDS)
+
+            println("Shard calculation took: $shardCreationTime")
+
+            result
+        } else {
+            calculateShards(
```

Comment:
testTargetsAlwaysRun support is missing. I think calculateShards should be removed